src/compiler/checker.ts

```diff
@@ -16215,9 +16215,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             const length = types.length;
             let i = 0;
             while (i < length) {
-                const startId = types[i].id;
+                const startId = types[i]?.id;

```

Comment:
This doesn't seem right; this array should never contain undefined. I would look for where the undefined entry is added as that would be the big, not this function which just iterates over the array of types.